test/NetTopologySuite.Tests.NUnit/LinearReferencing/LengthLocationMapTests.cs

```diff
@@ -0,0 +1,259 @@
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(30, length);
+        }
+
+        [Test]
+        public void TestMultiLineHoleLengthPosition50()
+        {
+            var line = Read("MULTILINESTRING((0 0, 0 50), (0 51, 0 100))");
+            var point = Read("POINT (0 50)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(50, length);
+        }
+
+        [Test]
+        public void TestMultiLineHoleLengthPosition60()
+        {
+            var line = Read("MULTILINESTRING((0 0, 0 50), (0 51, 0 100))");
+            var point = Read("POINT (0 60)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(59, length);
+        }
+
+        [Test]
+        public void TestMultiLineHoleLengthAtPosition100()
+        {
+            var line = Read("MULTILINESTRING((0 0, 0 50), (0 51, 0 100))");
+            var point = Read("POINT (0 100)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(99, length);
+        }
+
+        [Test]
+        public void TestMultiLineHoleLengthAtPosition101()
+        {
+            var line = Read("MULTILINESTRING((0 0, 0 50), (0 51, 0 100))");
+            var point = Read("POINT (0 101)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(99, length);
+        }
+
+        [Test]
+        public void TestMultiLineHoleLengthAtPosition0()
+        {
+            var line = Read("MULTILINESTRING((0 0, 0 50), (0 51, 0 100))");
+            var point = Read("POINT (0 0)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(0, length);
+        }
+
+        
+        [Test]
+        public void TestMultiLineHoleLengthAtPositionMinus1()
+        {
+            var line = Read("MULTILINESTRING((0 0, 0 50), (0 51, 0 100))");
+            var point = Read("POINT (0 -1)");
+
+            var loc = LocationIndexOfPoint.IndexOf(line, point.Coordinate);
+            double length = LengthLocationMap.GetLength(line, loc);
+            Assert.AreEqual(0, length);
+        }
+
+        private WKTReader _reader = new WKTReader();
+
+        protected Geometry Read(string wkt)
+        {
+            try
+            {
+                return _reader.Read(wkt);
+            }
+            catch (ParseException ex)
+            {
+                throw new ApplicationException("An exception occurred while reading the wkt", ex);
+            }
+        }
```

Comment:
Not necessary when deriving from GeometryTestCase

Reply:
Done